daemons/attrd/attrd_utils.c

```diff
@@ -166,20 +166,21 @@ attrd_value_needs_expansion(const char *value)
 int
 attrd_expand_value(const char *value, const char *old_value)
 {
-    int offset = 1;
-    int int_value = char2score(old_value);
+    int increment = 1;
+    int score = 0;
 
-    if (value[plus_plus_len + 1] != '+') {
-        const char *offset_s = value + (plus_plus_len + 2);
-
-        offset = char2score(offset_s);
+    if (pcmk_parse_score(old_value, &score, 0) != pcmk_rc_ok) {
+        return 0; // Original value is not a score
```

Comment:
Meanwhile, if the original value is `NULL`, we won't hit this; we'll treat it as 0 and increment it.

Reply:
That's intentional, otherwise someone couldn't unconditionally set value++. They'd have to query whether a value is currently set first and set an initial value if not. If we make them do that, they might as well increment it themselves.

Reply:
That's reasonable. It is odd though, because `pcmk__xe_set_score()` treats `value++` as a string literal when the original value is `NULL`. You've added a TODO, though I'm not sure how you plan to address it. Maybe by lining it up with this function.

Reply:
Just kicking that can down the road. Unless a user has an issue, I'd rather spend as little time as possible on expansion.